plugins/outputs/influxdb/client/http.go

```diff
@@ -233,15 +233,24 @@ func (c *httpClient) makeWriteRequest(
 		return nil, err
 	}
 	req.Header.Set("Content-Length", fmt.Sprint(contentLength))
-	// TODO
-	// if gzip {
-	// 	req.Header.Set("Content-Encoding", "gzip")
-	// }
+	if c.config.Gzip {
+		req.Header.Set("Content-Encoding", "gzip")
+	}
```

Comment:
Can you move all the Header code into makeRequest, I'm not sure why they should be separate.